### ISSUE TITLE
Some suggestions / improvements

### ISSUE THREAD

**Issue by p-realinho** [robinparisi/tingle#46]
Hey!

It's the first time I'm using Tingle, and I'm enjoying it so far!
Now, as I'm using it, I'm noticing some stuff that could (imo) be changed/improved/implemented.
(I apologize in advance if any of this things are already implemented, or in the 'TODO', or if I'm just misusing the plugin ... or, you know, if they're just dumb ahah )

- ~~ Have a seperate `modal.setTitle()` method ~~
- Have an option for showing/hiding the default close button (I'm using footer buttons to close the modal)
- Add `opacity` and `visibility` to the `.tingle-modal`'s `transition` (so that it fades out nicely)
- Have a way to keep the window scrollbar (I know they disappear for a reason, but I don't really like the little "translation" they cause when appearing and disappearing) (I wouldn't mind the content underneath the modal to be scrollable as a side-effect)
- Have a `.less` file too (because, vars)

I'd be more than happy to help out, so let me know if I should start submitting PRs left and right 😃

Edit: I just noticed there's already a PR for the `modal.setTitle()` method, so forget about that one.

**Comment by robinparisi** [comment]
Hi p-realinho,

Firstly, thanks for your suggestions :)

Regarding tingle, I’m trying to keep the plugin simple, extensible and clean. I don’t want to add a lot of features because many other plugins already do this very well.
 
For example, the modal's title can be done by setting content and some custom classes, and I think it’s a better way to do this for fitting every needs. It’s the same thing for the transition, the style is just a base that could/should be customized.

Concerning LESS, I explained [here](https://github.com/robinparisi/tingle/issues/25
) why I don’t want to add any preprocessor.

I know there are some issues with the scrollbar, and it’s really tricky to handle it properly because it relies on the browser and also depends on the OS but I’m definitely working on this.

Same for the close button and click outside behavior, I just want to think a little bit more about it before setting up a clean API.
 
I hope that I have been able to answer all your questions.
